cmd/argocd-application-controller/commands/argocd_application_controller.go

```diff
@@ -224,22 +223,21 @@ func getClusterFilter(kubeClient *kubernetes.Clientset, settingsMgr *settings.Se
 	}
 
 	if appControllerDeployment != nil && appControllerDeployment.Spec.Replicas != nil {
-		replicas = int(*appControllerDeployment.Spec.Replicas)
+		replicasCount = int(*appControllerDeployment.Spec.Replicas)
 	} else {
-		replicas = env.ParseNumFromEnv(common.EnvControllerReplicas, 0, 0, math.MaxInt32)
+		replicasCount = env.ParseNumFromEnv(common.EnvControllerReplicas, 0, 0, math.MaxInt32)
 	}
 
-	var clusterFilter func(cluster *v1alpha1.Cluster) bool
-	if replicas > 1 {
+	db := db.NewDB(settingsMgr.GetNamespace(), settingsMgr, kubeClient)
```

Comment:
In the previous implementation, that was the case, and I was able to delete the requirement of the DB here, but since the introduction of the `Deployment` based deployment, it seems that it is required now to manage properly scale up and scale down.
So I reverted the change and I pass the `DB` to the `ClusterSharding` .
I am moving it just before the call to `sharding.NewClusterSharding(db, shardNumber, replicasCount, shardingAlgorithm)` .